### PR TITLE
add Nullable proposal

### PR DESCRIPTION
**Pull request by i-walker**

### CODE REVIEW

File: arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/NullableSpec.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  "if expression short circuit" {
+    nullable {
+      val number = "s".length.bind()
+      val string = if (number != 1) {
+        number.toString()
+      } else {
+        null
+      }.bind()
+      string
+    } shouldBe null
```
Comment (nomisRev) [inline arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/NullableSpec.kt:112]: Can we also add a test checking if we can bind `Either<Nothing, A>`?

**Review by nomisRev — APPROVED** [review]
Looks good to me!

File: arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/NullableSpec.kt (hunk 1 above)
Comment (i-walker) [reply arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/NullableSpec.kt:113]: Added :)

File: arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/NullableSpec.kt [hunk 2]
```diff
@@ -111,4 +112,11 @@ class NullableSpec : StringSpec({
       string
     } shouldBe null
   }
+
+  "Either<Nothing, A> short circuit" {
```
Comment (nomisRev) [inline arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/NullableSpec.kt:116]:
```suggestion
"Either<Nothing, A> can be bind"
```

`Either<Nothing, A>` can never result in `Left` and thus also never short-circuit.

**Review by nomisRev — APPROVED** [review]

**Pull request merged**